pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -768,10 +783,73 @@ func (r *ReconcileBareMetalHost) actionManageReady(prov provisioner.Provisioner,
 	return r.manageHostPower(prov, info)
 }
 
-func (r *ReconcileBareMetalHost) saveStatus(host *metal3v1alpha1.BareMetalHost) error {
+func (r *ReconcileBareMetalHost) saveHostStatus(host *metal3v1alpha1.BareMetalHost) error {
+	if err := r.saveHostAnnotation(host); err != nil {
+		return err
+	}
+
+	//Refetch host again
+	err := r.client.Get(context.TODO(),
+		client.ObjectKey{
+			Name:      host.Name,
+			Namespace: host.Namespace,
+		},
+		host,
+	)
+
+	if err != nil {
+		return errors.Wrap(err, "Failed to update Status annotation")
+	}
+
 	t := metav1.Now()
 	host.Status.LastUpdated = &t
-	return r.client.Status().Update(context.TODO(), host)
+	errStatus := r.client.Status().Update(context.TODO(), host)
+	fmt.Printf("*********************************************** %v", errStatus)
+	return errStatus
+}
+
+func (r *ReconcileBareMetalHost) saveHostAnnotation(host *metal3v1alpha1.BareMetalHost) error {
+	//Repopulate annotation again
+	obj := r.getHostFromAnnotation(host)
+	if obj == nil {
+		return errors.New("Failed to get Host from Status annotation")
+	}
+
+	obj.Status.LastUpdated = host.Status.LastUpdated
+
+	if host.Status != obj.Status {
```

Comment:
you might need a deepEqual kind of check here

Reply:
Done